### PR TITLE
feat: nonstandard lro generation

### PR DESCRIPTION
**Pull request by viacheslav-rostovtsev**

### CODE REVIEW

File: gapic-generator/templates/default/service/client/_client.erb [hunk 1]
```diff
@@ -139,22 +134,17 @@ class <%= service.client_name %>
...
-<%- end -%>
-<%- if service.mixins? -%>
-  <%- service.mixins.each do |mixin| -%>
-  # @return [<%= mixin.client_class_name %>]
-  attr_reader :<%= mixin.client_var_name %>
+  attr_reader :<%= subclient.client_var_name %>
 
   <% end %>
-<% end %>
+  <% end %>
```
Comment (dazuma) [inline gapic-generator/templates/default/service/client/_client.erb:147]: We should change this and the previous line to `<%- end -%>` to eliminate extra whitespace.

File: gapic-common/lib/gapic/operation.rb [hunk 2]
```diff
@@ -86,33 +82,26 @@ class Operation
...
-    # Returns the server-assigned name of the operation, which is only unique within the same service that originally
-    # returns it. If you use the default HTTP mapping, the name should have the format of operations/some/unique/name.
+    # @return [Google::Longrunning::Operation] The wrapped grpc operation object.
     #
-    # @return [String] The name of the operation.
-    #
-    def name
-      @grpc_op.name
+    def grpc_op
+      @operation
```
[the remark this answers is not in the document]
Comment (viacheslav-rostovtsev) [reply gapic-common/lib/gapic/operation.rb:104]: fxed

File: gapic-generator/templates/default/service/client/_client.erb (hunk 1 above)
Comment (viacheslav-rostovtsev) [reply gapic-generator/templates/default/service/client/_client.erb:147]: :+1:

File: gapic-generator/lib/gapic/model/method/lro.rb [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+      #
+      # @!attribute [r] service_full_name
+      #   @return [String] Full grpc name of the provider service for this method's LRO
+      #   e.g. `google.example.LroProvider`
+      #
+      # @!attribute [r] operation_request_fields
+      #   @return [Map<String, String>] Map of the field names of this method's request message
+      #   that should be saved and later used when constructing the LRO polling request.
+      #   The values are the names of the fields in the polling request.
+      #   The semantics are `from` -> `to`, e.g. {`foo` => `bar`} means that
```
[the remark this answers is not in the document]
Comment (viacheslav-rostovtsev) [reply gapic-generator/lib/gapic/model/method/lro.rb:36]: :+1:

File: shared/output/cloud/compute_small/lib/google/cloud/compute/v1/addresses/rest/client.rb [hunk 4]
```diff
@@ -310,10 +329,10 @@ def delete request, options = nil
               #   @param region [::String]
               #     Name of the region for this request.
               # @yield [result, response] Access the result along with the Faraday response object
-              # @yieldparam result [::Google::Cloud::Compute::V1::Address]
+              # @yieldparam result [::Gapic::GenericLRO::Operation]
               # @yieldparam response [::Faraday::Response]
               #
-              # @return [::Google::Cloud::Compute::V1::Address]
+              # @return [::Gapic::GenericLRO::Operation]
```
Comment (dazuma) [inline shared/output/cloud/compute_small/lib/google/cloud/compute/v1/addresses/rest/client.rb:335]: How did this change happen? Is this going to cause a breaking change in the actual compute library?

Comment (viacheslav-rostovtsev) [reply shared/output/cloud/compute_small/lib/google/cloud/compute/v1/addresses/rest/client.rb]: Excellent catch, thank you!
This one was triggered by changes in the proto which were a copy-paste mistake I committed in a different PR. It appeared after a rebase and I didn't notice it when reviewing.
The mistake assigned a nonstandard LRO service to this method, even though it was not returning an `Operation` that said LRO service could poll.

- I have removed the copy-paste mistake
- And also I have added a check that will throw an error if a method gets assigned a NonstandardLRO service that cannot poll its return type.

**Review by dazuma — APPROVED** [review]

**Pull request merged**